File: haohaninfo/交易實務案例/73.R
#取得即時報價，詳細在技巧51
source("function.R")

#設定變數初始格式
inDesk<-0
outDesk<-0
#設定趨勢判斷時間
trendEndTime <- strptime('09:00:00.00','%H:%M:%OS')
trend <- NA

while(TRUE){
 #取得即時成交資訊
 Mdata<-GetMatchData(DataPath,Date)
 MatchTime <- strptime(Mdata[[1]][1],'%H:%M:%OS')
 MatchPrice <- as.numeric(Mdata[[1]][2])
 MatchQty <- as.numeric(Mdata[[1]][3])
 
 #取得上下五檔價資訊
 UpDn5data<-GetUpDn5Data(DataPath,Date) 
 Dn1Price<- as.numeric(UpDn5data[[1]][2])
 Up1Price<- as.numeric(UpDn5data[[1]][12])

 #進行內外盤判斷
 if ( MatchPrice >= Up1Price ){
  outDesk <- outDesk+MatchQty
 }else if( MatchPrice <= Dn1Price){
  inDesk <- inDesk+MatchQty
 }

 if( MatchTime >= trendEndTime & is.na(trend) ){
  if(outDesk > inDesk){
   trend <- 1
  }else if(outDesk < inDesk){
   trend <- (-1)
  }else{
   trend <- 0
  }

 }

 if(!is.na(trend)){
  print(trend)
 }
}

File: haohaninfo/即時範例/73.py
# -*- coding: UTF-8 -*-

#取得報價資訊，詳情請查看技巧51
execfile('function.py')

#定義判斷時間
trendTime=datetime.datetime.strptime('09:00:00.00',"%H:%M:%S.%f")
trend=0

#定義指標變數
OutDesk=0
InDesk=0

#取得成交資訊
for i in getMatch():		
 MatchInfo=i.split(',')
 MatchTime=datetime.datetime.strptime(MatchInfo[0],"%H:%M:%S.%f")
 MatchPrcie=int(MatchInfo[1])
 MatchQty=int(MatchInfo[2])
 UpDn5Info=getLastUpDn5()
 Dn1Price=int(UpDn5Info[1])
 Up1Price=int(UpDn5Info[11])

 if MatchPrcie>=Up1Price:
  OutDesk+=MatchQty
 if MatchPrcie<=Dn1Price:
  InDesk+=MatchQty

 #趨勢判斷
 if MatchTime >= trendTime:
  if OutDesk>InDesk:
   trend+=1
  elif OutDesk<InDesk:
   trend-=1 
  break
 
 print MatchInfo[0],"OutDesk",OutDesk,"InDesk",InDesk

print "Trend",trend